src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -729,13 +769,19 @@ static int dhcp_server_relay_message(sd_dhcp_server *server, DHCPMessage *messag
                 if (message->giaddr == 0)
                         message->giaddr = server->address;
 
+                if (server->agent_circuit_id || server->agent_remote_id) {
+                        r = append_agent_information_option(server, message, opt_length + sizeof(DHCPMessage), buflen - sizeof(DHCPMessage));
```

Comment:
Why do you add and subtract `sizeof(DHCPMessage)` so many times?? That's complicated and seems meaningless...

Reply:
Ok, cleaned this up a bit. Though in general I do not see the way how to make it cleaner.